src/modules/map/components/LocationPopup.vue

```diff
@@ -143,7 +144,14 @@ export default {
             return this.clickInfo && this.clickInfo.clickType === ClickType.RIGHT_CLICK
         },
         shareLinkUrl: function () {
-            return `${window.location}&crosshair=marker`
+            let [lon, lat] = this.reprojectClickCoordinates('EPSG:4326')
+            let query = {
+                ...this.$route.query,
+                crosshair: 'marker',
+                lat,
+                lon,
+            }
+            return `${location.origin}/#/map?${stringifyQuery(query)}`
```

Comment:
Just a newbea javascript question, why not using `window.location.origin` here ?

Reply:
I'll see what I can do about the QR code 😃 

Concerning the use of `window`. There is no definitive answer. Generally speaking you never need to write `window` as every variable that is not defined will be set/looked up on the global scope (i.e. window).

So if you write `foo = 1` somewhere in your code (without having defined `foo` somewhere), you will set `window.foo`. The same goes for reading variables.

So. Whether or not you write `window` comes down to convention and/or preference. I generally avoid `window` unless the code is not clear otherwise. In essence, I'll write window for custom variables on the global scope but not for existing properties from the browser APIs (location, document, console, localStorage, ...). But I make exceptions in cases like `window.innerHeight`, etc.

The MDN pages of the window properties sometimes say what the convention is. But I'm not really going after that anymore. But it's a good start until you have your own preferences.

Reply:
Thanks for the explaination :-)